pkg/debug/info.go

```diff
@@ -0,0 +1,140 @@
+package debug
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/golang/glog"
+	"github.com/openshift/odo/pkg/occlient"
+	"github.com/openshift/odo/pkg/testingutil/filesystem"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"net"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+type OdoDebugFile struct {
+	metav1.TypeMeta
+	DebugProcessId int
+	ProjectName    string
+	AppName        string
+	ComponentName  string
+	RemotePort     int
+	LocalPort      int
+}
+
+// GetDebugInfoFilePath gets the file path of the debug info file
+func GetDebugInfoFilePath(client *occlient.Client, componentName, appName string) string {
+	tempDir := os.TempDir()
+	debugFilePrefix := "odo-debug.json"
+	s := []string{client.Namespace, appName, componentName, debugFilePrefix}
+	debugFileName := strings.Join(s, "-")
+	return filepath.Join(tempDir, debugFileName)
+}
+
+func CreateDebugInfoFile(f *DefaultPortForwarder, portPair string) error {
+	return createDebugInfoFile(f, portPair, filesystem.DefaultFs{})
+}
+
+// createDebugInfoFile creates a file in the temp directory with information regarding the debugging session of a component
+func createDebugInfoFile(f *DefaultPortForwarder, portPair string, fs filesystem.Filesystem) error {
+	portPairs := strings.Split(portPair, ":")
+	if len(portPairs) > 2 || len(portPairs) < 2 {
```

Comment:
shouldn't we just check if the length is just `!=2`?

Reply:
fixed